charms/worker/k8s/src/token_distributor.py

```diff
@@ -164,24 +164,24 @@ def _create_cos_token(self, name: str, _):
             username=f"system:cos:{name}", groups=["system:cos"]
         )
 
-    def _revoke_cluster_token(self, name: str, _):
+    def _revoke_cluster_token(self, name: str, ignore_errors: bool):
         """Remove a cluster token.
 
         Args:
             name (str): The name of the node.
+            ignore_errors (bool): Whether or not errors can be ignored
 
         Raises:
             K8sdConnectionError: reraises cluster token revoke failures
         """
         try:
             self.api_manager.remove_node(name)
-        except (K8sdConnectionError, InvalidResponseError):
-            if name == self.node_name:
-                # I'm unclustering myself
+        except (K8sdConnectionError, InvalidResponseError) as e:
+            if ignore_errors:
                 # Let's just ignore some of these expected errors:
                 # "Remote end closed connection without response"
                 # "Failed to check if node is control-plane"
-                log.exception("Unclustering ones self can expect an error")
+                log.warning("Unclustering self leads to an expected error: %s", e)
```

Comment:
Having a stack-trace here wasn't useful debug.  
Oo... just noticed this warning isn't quite right either.  There may be other reasons we are ignoring errors.

Reply:
Fixed